src/components/component_card.js

```diff
@@ -5,7 +5,10 @@ import * as actions from '../actions';
 class CardComponent extends Component {
   constructor(props) {
     super(props);
-
+    // move this whole state to a reducer and trigger actions accordingly
+    // this way you will keep the logic of the application only to
+    // the level of the reducers and not the component level.
+    
```

Comment:
You should consider what does and does not need to be in the state (e.g. what changes?)

For anything that does change, try and move this into the main application state so it could be handled by any component, not just this one - based on the contents of the store. This makes the whole thing much more generic and less likely to break.

There should be nothing left in this individual component state when you are done.